[Bug]: `TimeIntervalsInterface` conversion option `column_name_mapping` cannot be used from a Converter

### What happened?

Using `column_name_mapping` in `conversion_options` is not possible when the `TimeIntervalsInterface` is used from a converter. The conversion specification schema expects it to be `string` type and `ValidationError` is raised when dict is used for `column_name_mapping`. 

Example:
` jsonschema.exceptions.ValidationError: {'condition': 'cond'} is not of type 'string'`

### Steps to Reproduce

```python
class TestConverter(NWBConverter):
        data_interface_classes = dict(TestRecording=ExcelTimeIntervalsInterface)

    converter = TestConverter(source_data=dict(TestRecording=dict(file_path=trials_xls_path)))
    metadata = converter.get_metadata()
    metadata["NWBFile"].update(session_start_time=datetime.now().astimezone())
    converter.run_conversion(
        nwbfile_path="test/test.nwb",
        overwrite=True,
        metadata=metadata,
        conversion_options=dict(
            TestRecording=dict(
                column_name_mapping=dict(condition="cond")),
    ))
```


### Traceback

```python
>       converter.run_conversion(
            nwbfile_path="test/test.nwb",
            overwrite=True,
            metadata=metadata,
            conversion_options=dict(
                TestRecording=dict(
                    column_name_mapping=dict(condition="cond")),
        ))

test_time_intervals.py:65: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/neuroconv/nwbconverter.py:173: in run_conversion
    self.validate_conversion_options(conversion_options=conversion_options_to_run)
../../src/neuroconv/nwbconverter.py:123: in validate_conversion_options
    validate(instance=conversion_options, schema=self.get_conversion_options_schema())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

instance = {'TestRecording': {'column_name_mapping': {'condition': 'cond'}}}
schema = {'$id': 'conversion_options.schema.json', '$schema': 'http://json-schema.org/draft-07/schema#', 'additionalProperties': False, 'description': 'Schema for the conversion options', ...}
cls = <class 'jsonschema.validators.Draft7Validator'>, args = (), kwargs = {}
validator = Draft7Validator(schema={'$id': 'conversion_o...s.schema.json', '$schema': 'http://json-...ft-07/schema#', 'additionalProperties': False, 'description': 'Schema for t...rsion options', ...}, format_checker=None)
error = <ValidationError: "{'condition': 'cond'} is not of type 'string'">

    def validate(instance, schema, cls=None, *args, **kwargs):
        """
        Validate an instance under the given schema.
    
            >>> validate([2, 3, 4], {"maxItems": 2})
            Traceback (most recent call last):
                ...
            ValidationError: [2, 3, 4] is too long
    
        :func:`validate` will first verify that the provided schema is
        itself valid, since not doing so can lead to less obvious error
        messages and fail in less obvious or consistent ways.
    
        If you know you have a valid schema already, especially
        if you intend to validate multiple instances with
        the same schema, you likely would prefer using the
        `jsonschema.protocols.Validator.validate` method directly on a
        specific validator (e.g. ``Draft20212Validator.validate``).
    
    
        Arguments:
    
            instance:
    
                The instance to validate
    
            schema:
    
                The schema to validate with
    
            cls (jsonschema.protocols.Validator):
    
                The class that will be used to validate the instance.
    
        If the ``cls`` argument is not provided, two things will happen
        in accordance with the specification. First, if the schema has a
        :kw:`$schema` keyword containing a known meta-schema [#]_ then the
        proper validator will be used. The specification recommends that
        all schemas contain :kw:`$schema` properties for this reason. If no
        :kw:`$schema` property is found, the default validator class is the
        latest released draft.
    
        Any other provided positional and keyword arguments will be passed
        on when instantiating the ``cls``.
    
        Raises:
    
            `jsonschema.exceptions.ValidationError`:
    
                if the instance is invalid
    
            `jsonschema.exceptions.SchemaError`:
    
                if the schema itself is invalid
    
        .. rubric:: Footnotes
        .. [#] known by a validator registered with
            `jsonschema.validators.validates`
        """
        if cls is None:
            cls = validator_for(schema)
    
        cls.check_schema(schema)
        validator = cls(schema, *args, **kwargs)
        error = exceptions.best_match(validator.iter_errors(instance))
        if error is not None:
>           raise error
E           jsonschema.exceptions.ValidationError: {'condition': 'cond'} is not of type 'string'
E           
E           Failed validating 'type' in schema['properties']['TestRecording']['properties']['column_name_mapping']:
E               {'type': 'string'}
E           
E           On instance['TestRecording']['column_name_mapping']:
E               {'condition': 'cond'}

/Users/weian/opt/anaconda3/envs/neuroconv39/lib/python3.9/site-packages/jsonschema/validators.py:1116: ValidationError
```


### Operating System

macOS

### Python Executable

Python

### Python Version

3.9

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@
 * Changed `SpikeGLXRecordingInterface` to accept either the AP or LF bands as file paths. Each will automatically set the correseponding `es_key` and corresponding metadata for each band or probe. [PR #298](https://github.com/catalystneuro/neuroconv/pull/298)
 * The `OpenEphysRecordingInterface` redirects to `OpenEphysLegacyRecordingInterface` for legacy format files instead of raising NotImplementedError. [PR #349](https://github.com/catalystneuro/neuroconv/pull/349)
 * Added an `interfaces_by_category` lookup table to `neuroconv.datainterfaces` to make searching for interfaces by modality and format easier. [PR #352](https://github.com/catalystneuro/neuroconv/pull/352)
+* `neuroconv.utils.jsonschema.get_schema_from_method_signature` can now support the `Dict[str, str]` typehint, which allows `DataInterface.__init__` and `.run_conversion` to handle dictionary arguments. [#360](https://github.com/catalystneuro/neuroconv/pull/360)
 
 ### Testing
 * The tests for `automatic_dandi_upload` now follow up-to-date DANDI validation rules for file name conventions. [PR #310](https://github.com/catalystneuro/neuroconv/pull/310)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,7 +19,7 @@
 * Changed `SpikeGLXRecordingInterface` to accept either the AP or LF bands as file paths. Each will automatically set the correseponding `es_key` and corresponding metadata for each band or probe. [PR #298](https://github.com/catalystneuro/neuroconv/pull/298)
 * The `OpenEphysRecordingInterface` redirects to `OpenEphysLegacyRecordingInterface` for legacy format files instead of raising NotImplementedError. [PR #349](https://github.com/catalystneuro/neuroconv/pull/349)
 * Added an `interfaces_by_category` lookup table to `neuroconv.datainterfaces` to make searching for interfaces by modality and format easier. [PR #352](https://github.com/catalystneuro/neuroconv/pull/352)
-* `neuroconv.utils.jsonschema.get_schema_from_method_signature` can now support the `Dict[str, str]` typehint, which allows `DataInterface.__init__` and `.run_conversion` to handle dictionary arguments. [#360](https://github.com/catalystneuro/neuroconv/pull/360)
+* `neuroconv.utils.jsonschema.get_schema_from_method_signature` can now support the `Dict[str, str]` typehint, which allows `DataInterface.__init__` and `.run_conversion` to handle dictionary arguments. [PR #360](https://github.com/catalystneuro/neuroconv/pull/360)
 
 ### Testing
 * The tests for `automatic_dandi_upload` now follow up-to-date DANDI validation rules for file name conventions. [PR #310](https://github.com/catalystneuro/neuroconv/pull/310)

--- a/src/neuroconv/utils/json_schema.py
+++ b/src/neuroconv/utils/json_schema.py
@@ -2,7 +2,7 @@ import collections.abc
 import inspect
 import json
 from datetime import datetime
-from typing import Literal, Dict
+from typing import Dict, Literal
 
 import numpy as np
 import pynwb

--- a/src/neuroconv/utils/json_schema.py
+++ b/src/neuroconv/utils/json_schema.py
@@ -2,7 +2,7 @@ import collections.abc
 import inspect
 import json
 from datetime import datetime
-from typing import Literal
+from typing import Literal, Dict
 
 import numpy as np
 import pynwb
@@ -84,6 +84,12 @@ def get_schema_from_method_signature(class_method: classmethod, exclude: list = 
         if param.annotation:
             if getattr(param.annotation, "__origin__", None) == Literal:
                 args_spec[param_name]["enum"] = list(param.annotation.__args__)
+            elif getattr(param.annotation, "__origin__", None) == dict:
+                args_spec[param_name] = dict(type="object")
+                if param.annotation.__args__ == (str, str):
+                    args_spec[param_name].update(additionalProperties={"^.*$": dict(type="string")})
+                else:
+                    args_spec[param_name].update(additionalProperties=True)
             elif hasattr(param.annotation, "__args__"):  # Annotation has __args__ if it was made by typing.Union
                 args = param.annotation.__args__
                 valid_args = [x.__name__ in annotation_json_type_map for x in args]

--- a/tests/test_minimal/test_utils/test_json_schema_utils.py
+++ b/tests/test_minimal/test_utils/test_json_schema_utils.py
@@ -2,7 +2,7 @@ import json
 import os
 from copy import deepcopy
 from pathlib import Path
-from typing import Union
+from typing import Union, Dict
 
 from neuroconv.utils import (
     dict_deep_update,
@@ -35,7 +35,7 @@ def sort_item(item):
 
 def test_get_schema_from_method_signature():
     class A:
-        def __init__(self, a: int, b: float, c: Union[Path, str], d: bool, e: str = "hi"):
+        def __init__(self, a: int, b: float, c: Union[Path, str], d: bool, e: str = "hi", f: Dict[str, str] = None):
             pass
 
     schema = get_schema_from_method_signature(A.__init__)
@@ -48,6 +48,7 @@ def test_get_schema_from_method_signature():
             c=dict(type="string"),
             d=dict(type="boolean"),
             e=dict(default="hi", type="string"),
+            f=dict(type="object", additionalProperties={'^.*$': dict(type='string')})
         ),
         required=[
             "a",

--- a/tests/test_minimal/test_utils/test_json_schema_utils.py
+++ b/tests/test_minimal/test_utils/test_json_schema_utils.py
@@ -2,7 +2,7 @@ import json
 import os
 from copy import deepcopy
 from pathlib import Path
-from typing import Union, Dict
+from typing import Dict, Union
 
 from neuroconv.utils import (
     dict_deep_update,
@@ -48,7 +48,7 @@ def test_get_schema_from_method_signature():
             c=dict(type="string"),
             d=dict(type="boolean"),
             e=dict(default="hi", type="string"),
-            f=dict(type="object", additionalProperties={'^.*$': dict(type='string')})
+            f=dict(type="object", additionalProperties={"^.*$": dict(type="string")}),
         ),
         required=[
             "a",

